DUI3-DX/Converters/Revit/Speckle.Converters.RevitShared/ToHost/TopLevel/ModelCurveToSpeckleTopLevelConverter.cs

```diff
@@ -0,0 +1,110 @@
+using Objects;
+using Speckle.Converters.Common;
+using Speckle.Converters.Common.Objects;
+using Speckle.Converters.RevitShared.Helpers;
+using System.Collections;
+
+namespace Speckle.Converters.RevitShared.ToSpeckle;
+
+[NameAndRankValue(nameof(SOBR.Curve.ModelCurve), 0)]
+public class ModelCurveToHostTopLevelConverter : BaseTopLevelConverterToHost<SOBR.Curve.ModelCurve, DB.ModelCurve>
+{
+  private readonly ITypedConverter<ICurve, DB.CurveArray> _curveConverter;
+  private readonly IRevitConversionContextStack _contextStack;
+
+  public ModelCurveToHostTopLevelConverter(
+    ITypedConverter<ICurve, DB.CurveArray> curveConverter,
+    IRevitConversionContextStack conversionContext
+  )
+  {
+    _curveConverter = curveConverter;
+    _contextStack = conversionContext;
+  }
+
+  public override DB.ModelCurve Convert(SOBR.Curve.ModelCurve target) =>
+    ModelCurvesFromEnumerator(_curveConverter.Convert(target.baseCurve).GetEnumerator(), target.baseCurve);
+
+  private DB.ModelCurve ModelCurvesFromEnumerator(IEnumerator curveEnum, ICurve speckleLine)
+  {
+    DB.ModelCurve? revitCurve = null;
+    while (curveEnum.MoveNext() && curveEnum.Current != null)
+    {
+      var curve = (DB.Curve)curveEnum.Current;
+      // Curves must be bound in order to be valid model curves
+      if (!curve.IsBound)
+      {
+        curve.MakeBound(speckleLine.domain.start ?? 0, speckleLine.domain.end ?? Math.PI * 2);
+      }
+
+      if (_contextStack.Current.Document.IsFamilyDocument)
+      {
+        revitCurve = _contextStack.Current.Document.FamilyCreate.NewModelCurve(
+          curve,
+          NewSketchPlaneFromCurve(curve, _contextStack.Current.Document)
+        );
+      }
+      else
+      {
+        revitCurve = _contextStack.Current.Document.Create.NewModelCurve(
+          curve,
+          NewSketchPlaneFromCurve(curve, _contextStack.Current.Document)
+        );
+      }
+    }
+
+    return revitCurve.NotNull();
```

Comment:
So we're saying, we expected to make a curve but this didn't happen. I think that's probably but I see we might make many curves, is it the case that we expect to make at least one curve?